pkg/ast/spl/spl.peg

```diff
@@ -2030,6 +2037,112 @@ Subseconds <- ("us" / "ms" / "cs" / "ds") {
     return timeUnit, nil
 }
 
+// Returns *structs.QueryAggregators
+TransactionBlock <- PIPE CMD_TRANSACTION txnOptions:TransactionOptions {
+    queryAgg := &structs.QueryAggregators{
+        PipeCommandType: structs.TransactionType,
+        TransactionArguments: txnOptions.(*structs.TransactionArguments),
+    }
+    return queryAgg, nil
+}
+
+// The List of fields should Always be given at the last.
+// The TransactionDefinitionOptionsList will consist of all the Transaction Options
+TransactionOptions <- txnOptions:TransactionDefinitionOptionsList?  {
+
+    transactionRequest := &structs.TransactionArguments{
+        Fields:     nil,
+        StartsWith: "",
+        EndsWith:   "",
+    }
+     
+    if txnOptions != nil {
+        txnArgs := txnOptions.(*TxnArgs)
+        transactionRequest.Fields = txnArgs.fields
+        transactionRequest.StartsWith = txnArgs.startsWith
+        transactionRequest.EndsWith = txnArgs.endsWith
+    }
+
+    return transactionRequest, nil
+}
+
+
+TransactionDefinitionOptionsList <- first:TransactionDefinitionOption rest:(SPACE TransactionDefinitionOption)* {
+
+    restSlice := rest.([]any)
+    txnArgs := &TxnArgs{
+        argOption: "txn-definition",
+        startsWith: "",
+        endsWith: "",
+    }
+
+    numArgs := 1 + len(restSlice)
+
+    for i := 0; i < numArgs; i++ {
+        var txnArg *TxnArgs
+        if i == 0 {
+            txnArg = first.(*TxnArgs)
+        } else {
+            separatorAndArg := restSlice[i - 1].([]any)
+            txnArg = separatorAndArg[1].(*TxnArgs)
+        }
+        argOption := txnArg.argOption
+
+        switch argOption {
+            case "fields":
+                txnArgs.fields = txnArg.fields
+            case "startswith": 
+                txnArgs.startsWith = txnArg.startsWith
+            case "endswith":
+                txnArgs.endsWith = txnArg.endsWith
+            default: 
+                return nil, fmt.Errorf("Not a Valid Transaction Argument option")
+        }
+    }
+
+    return txnArgs, nil
+}
+
+TransactionDefinitionOption <- option:(SpaceSeparatedFieldNameList / StartsWithOption / EndsWithOption) {
+    return option, nil
+}
+
+// Parses one or more FieldNames separate by a space
+// Returns a slice containing all the FieldNames
+SpaceSeparatedFieldNameList <- first:FieldName !(EQUAL) rest:(SPACE FieldName !(EQUAL))* {
+     var fields []string
+    fields = append(fields, first.(string))
+    for _, r := range rest.([]any) {
+        // Extracting the field name from the tuple (SPACE, FieldName)
+        fields = append(fields, r.([]any)[1].(string))
+    }
+
+    txnArg := &TxnArgs{
+        argOption: "fields",
+        fields: fields,
+    }
+
+    return txnArg, nil
+}
+
+StartsWithOption <- "startswith" EQUAL str:QuotedString {
+    txnArg := &TxnArgs{
+        argOption: "startswith",
+        startsWith: removeQuotes(str.(string)),
+    }
+    return txnArg, nil
+}
+
+EndsWithOption <- "endswith" EQUAL str:QuotedString {
```

Comment:
The Splunk documentation says the startswith and endswith values are "a search or eval filtering expression". I think for the initial PR you can keep these as they are, but eventually we'll need to support things like `startswith=(apple OR "banana")`. We should be able to reuse some of the existing rules in this file to handle that parsing.